src/graph/impl/KokkosGraph_GraphColor_impl.hpp

```diff
@@ -2379,6 +2379,234 @@ class GraphColor_VB:public GraphColor <HandleType,lno_row_view_t_,lno_nnz_view_t
+      Kokkos::parallel_for("Deterministic Coloring: color nodes in frontier", frontierSize(),
+                           KOKKOS_LAMBDA(const size_type frontierIdx) {
+
+                             size_type frontierNode = frontier(frontierIdx);
+                             int* bannedColors = new int[maxColors];
+                             for(size_type colorIdx= 0; colorIdx < maxColors; ++colorIdx) {
+                               bannedColors[colorIdx] = 0;
+                             }
+
+                             // Loop over neighbors, find banned colors, decrement dependency and update newFrontier
+                             for(size_type neigh = this->xadj(frontierNode); neigh < this->xadj(frontierNode + 1); ++neigh) {
+                               bannedColors[colors(this->adj(neigh))] = 1;
+                               dependency(this->adj(neigh)) = dependency(this->adj(neigh)) - 1;
+                               if(dependency(this->adj(neigh)) == 0) {
+                                 const size_type newFrontierIdx
+                                   = Kokkos::atomic_fetch_add(&newFrontierSize(), 1);
+                                 newFrontier(newFrontierIdx) = this->adj(neigh);
+                               }
+                             } // Loop over neighbors
+
+                             for(size_type color = 1; color < maxColors; ++color) {
+                               if(bannedColors[color] == 0) {
+                                 colors(frontierNode) = color;
+                                 break;
+                               }
+                             } // Loop over banned colors
+                             delete bannedColors;
+                           }); // Loop over current frontier
+    } // while newFrontierSize
+
+    print1DView("Colors", this->nv, colors);
+
+  } // color_graph()
+
+  void print1DView(const std::string label, const size_type frontierSize,
+                   const nnz_lno_temp_work_view_t frontier) {
+    std::cout << label << ": {";
+    for(size_type frontierIdx = 0; frontierIdx < frontierSize-1; ++frontierIdx) {
+      std::cout << frontier(frontierIdx) << " - ";
+    }
+    std::cout << frontier(frontierSize - 1) << "}" << std::endl;
+  } // printFrontier
+
+  template <class score_type, class max_type, class execution_space>
+  struct functorScoreCalcution {
+    typedef typename Kokkos::Experimental::Max<max_type, execution_space>::value_type valueType;
+    score_type score_;
+    const_lno_row_view_t numNeighbors_;
+
+    functorScoreCalcution(score_type score, const_lno_row_view_t numNeighbors)
+      : score_(score), numNeighbors_(numNeighbors) {}
+
+    KOKKOS_INLINE_FUNCTION
+    void operator() (const int i, valueType &update) const {
+      score_(i) = numNeighbors_(i + 1) - numNeighbors_(i);
+      update = ( (valueType) score_(i) < update ? update : (valueType) score_(i) );
+    }
+  }; // functorScoreCalcution()
+
```

Comment:
This appears like a good baseline to start with and optimize (except the new for bannedcolors above).